crates/hyperswitch_connectors/src/connectors/elavon/transformers.rs

```diff
@@ -1,228 +1,607 @@
-                refund_status: enums::RefundStatus::from(item.response.status),
+            status,
+            response,
+            ..item.data
+        })
+    }
+}
+impl TryFrom<RefundsResponseRouterData<Execute, ElavonPaymentsResponse>>
+    for RefundsRouterData<Execute>
+{
+    type Error = error_stack::Report<errors::ConnectorError>;
+    fn try_from(
+        item: RefundsResponseRouterData<Execute, ElavonPaymentsResponse>,
+    ) -> Result<Self, Self::Error> {
+        let status = enums::RefundStatus::from(&item.response);
+        let response = match &item.response {
+            ElavonPaymentsResponse::Error(error) => Err(ErrorResponse {
+                code: error.error_code.clone(),
+                message: error.error_message.clone(),
+                reason: Some(error.error_message.clone()),
+                attempt_status: None,
+                connector_transaction_id: None,
+                status_code: item.http_code,
             }),
+            ElavonPaymentsResponse::Success(response) => {
+                if status == enums::RefundStatus::Failure {
+                    Err(ErrorResponse {
+                        code: response.ssl_result_message.clone(),
+                        message: response.ssl_result_message.clone(),
+                        reason: Some(response.ssl_result_message.clone()),
+                        attempt_status: None,
+                        connector_transaction_id: None,
+                        status_code: item.http_code,
+                    })
+                } else {
+                    Ok(RefundsResponseData {
+                        connector_refund_id: response.ssl_txn_id.clone(),
+                        refund_status: enums::RefundStatus::from(&item.response),
+                    })
+                }
+            }
+        };
+        Ok(Self {
+            response,
             ..item.data
         })
     }
 }
 
-//TODO: Fill the struct with respective fields
-#[derive(Default, Debug, Serialize, Deserialize, PartialEq)]
-pub struct ElavonErrorResponse {
-    pub status_code: u16,
-    pub code: String,
-    pub message: String,
-    pub reason: Option<String>,
+trait ElavonResponseValidator {
+    fn is_successful(&self) -> bool;
+}
+impl ElavonResponseValidator for ElavonPaymentsResponse {
+    fn is_successful(&self) -> bool {
+        match self {
+            Self::Error(_) => false,
+            Self::Success(response) => {
+                if !matches!(response.ssl_result, SslResult::ImportedBatchFile) {
+                    return false;
+                }
+
+                matches!(
+                    response.ssl_issuer_response,
+                    None | Some(SslIssuerResponse::Success)
+                        | Some(SslIssuerResponse::NoDeclineReason)
+                        | Some(SslIssuerResponse::PartialSuccess10)
+                        | Some(SslIssuerResponse::PartialSuccess11)
```

Comment:
Should we consider transaction as successful if we get `PartialSuccess10` or `PartialSuccess10` in response?

Reply:
ideally not , In their docs they say Success is 00 only
```
Original issuer response
The issuer response returned during the authorization attempt.
An ssl_issuer_response = 00 indicates success.
An ssl_issuer_response = 85 indicates no reason to decline.
An ssl_issuer_response = 10 or 11 indicates partial success.
An ssl_issuer_response not equal to 00, 10, 11, or 85 indicates a decline or failure.
```